### ISSUE TITLE
Upgrade from 1.9.6 to 2.0.1 causes pyproj.exceptions.ProjError: Error creating CRS to CRS

### ISSUE THREAD

**Issue by KAMI911** [pyproj4/pyproj#191]
I mainly using and testing pyproj under Linux /Ubuntu 18.04 64 bit/ and Travis CI environment.

While I upgraded to latest 2.0.1 version from 1.9.6 version I hit into this problem:
https://travis-ci.org/KAMI911/lactransformer/jobs/505569736

ERROR: test_las_transformation_las_to_EOV2009 (__main__.TestLasTransformation_from_WGS84geo)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_lactransformer.py", line 388, in test_las_transformation_las_to_EOV2009
    self.las_data.TransformPointCloud()
  File "/home/travis/build/KAMI911/lactransformer/lactransformer/libs/LasPyConverter.py", line 133, in TransformPointCloud
    self.__DestinationOpenedFile.x, self.__DestinationOpenedFile.y, self.__DestinationOpenedFile.z = transform(self.__SourceProj, self.__DestinationProj, self.__SourceOpenedFile.x, self.__SourceOpenedFile.y, self.__SourceOpenedFile.z)
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/pyproj/__init__.py", line 507, in transform
    _proj._transform(p1, p2, inx, iny, inz)
  File "pyproj/_proj.pyx", line 302, in pyproj._proj._transform
  File "pyproj/_proj.pyx", line 276, in pyproj._proj.TransProj.__init__
pyproj.exceptions.ProjError: Error creating CRS to CRS.

When I switch back to 1.9.6 the problems are gone:
https://travis-ci.org/KAMI911/lactransformer/jobs/505579144

Some printings that may help:

(pyproj.Proj('+proj=geocent +ellps=WGS84 +datum=WGS84 +units=m +no_defs ', preserve_units=True), pyproj.Proj('+proj=somerc +lat_0=47.14439372222222 +lon_0=19.04857177777778 +k_0=0.99993 +x_0=650000 +y_0=200000 +ellps=GRS67 +nadgrids=/home/travis/build/KAMI911/lactransformer/lactransformer/grid/etrs2eov_notowgs.gsb +geoidgrids=/home/travis/build/KAMI911/lactransformer/lactransformer/grid/geoid_eht.gtx +units=m +no_defs ', preserve_units=True), array([4072352.038, 4072350.845, 4072352.694, ..., 4071665.054,
       4071689.017, 4071695.836]), array([1406781.664, 1406783.521, 1406781.519, ..., 1407165.48 ,
       1407119.194, 1407133.864]), array([4687388.295, 4687382.027, 4687385.287, ..., 4687839.668,
       4687832.19 , 4687823.562]))
(pyproj.Proj('+proj=geocent +ellps=WGS84 +datum=WGS84 +units=m +no_defs ', preserve_units=True), pyproj.Proj('+proj=somerc +lat_0=47.14439372222222 +lon_0=19.04857177777778 +k_0=0.99993 +x_0=650000 +y_0=200000 +ellps=GRS67 +nadgrids=/home/travis/build/KAMI911/lactransformer/lactransformer/grid/etrs2eov_notowgs.gsb +geoidgrids=/home/travis/build/KAMI911/lactransformer/lactransformer/grid/geoid_eht2014.gtx +units=m +no_defs ', preserve_units=True), array([4072352.038, 4072350.845, 4072352.694, ..., 4071665.054,
       4071689.017, 4071695.836]), array([1406781.664, 1406783.521, 1406781.519, ..., 1407165.48 ,
       1407119.194, 1407133.864]), array([4687388.295, 4687382.027, 4687385.287, ..., 4687839.668,
       4687832.19 , 4687823.562]))

My project's source:

https://github.com/KAMI911/lactransformer/tree/master/lactransformer

Converter class, where I hit to error:
https://github.com/KAMI911/lactransformer/blob/master/lactransformer/libs/LasPyConverter.py#L18

Maybe these Proj instance works differently?
self.__SourceProj = Proj(self.__SourceProjectionString)
self.__DestinationProj = Proj(self.__DestinationProjectionString)

and causes this exception:

Traceback (most recent call last):
  File "test_lactransformer.py", line 388, in test_las_transformation_las_to_EOV2009
    self.las_data.TransformPointCloud()
  File "/home/travis/build/KAMI911/lactransformer/lactransformer/libs/LasPyConverter.py", line 133, in TransformPointCloud
    self.__DestinationOpenedFile.x, self.__DestinationOpenedFile.y, self.__DestinationOpenedFile.z = transform(self.__SourceProj, self.__DestinationProj, self.__SourceOpenedFile.x, self.__SourceOpenedFile.y, self.__SourceOpenedFile.z)
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/pyproj/__init__.py", line 507, in transform
    _proj._transform(p1, p2, inx, iny, inz)
  File "pyproj/_proj.pyx", line 302, in pyproj._proj._transform
  File "pyproj/_proj.pyx", line 276, in pyproj._proj.TransProj.__init__
pyproj.exceptions.ProjError: Error creating CRS to CRS.

Is this a bug, o
r did I wrong something? From the documentation I can not see if I have to use `Proj` class differently.

**Comment by snowman2** [comment]
I confirmed this & made a fix in PR #191.

**Comment by KAMI911** [comment]
Still not working with 2.1.0.
I have strange coordinates in degree, but no pyproj.exceptions.ProjError: Error creating CRS to CRS. exception.

**Comment by snowman2** [comment]
Glad to hear the error is gone!

@KAMI911, the default for input/output for geographic projections is degrees. If you would prefer radians, you can use the `radians=True` flag on the `transform`.

**Comment by KAMI911** [comment]
I am generally use meter based projections like:
                   'WGS84geo': '+proj=geocent +ellps=WGS84 +datum=WGS84 +units=m +no_defs',
                   'EOV': '+proj=somerc +lat_0=47.14439372222222 +lon_0=19.04857177777778 +k_0=0.99993 +x_0=650000 +y_0=200000 +ellps=GRS67 +units=m +no_defs',
                   'EOVc': '+proj=somerc +lat_0=47.14439372222222 +lon_0=19.04857177777778 +k_0=0.99993 +x_0=650000 +y_0=200000 +ellps=GRS67 +nadgrids=' + nadgrids_EOV2014 + ' +geoidgrids=' + geoidgrids_EOV2014 + ' +units=m +no_defs',
                   'EOV2014': '+proj=somerc +lat_0=47.14439372222222 +lon_0=19.04857177777778 +k_0=0.99993 +x_0=650000 +y_0=200000 +ellps=GRS67 +nadgrids=' + nadgrids_EOV2014 + ' +geoidgrids=' + geoidgrids_EOV2014 + ' +units=m +no_defs',
                   'EOV2014fine': '+proj=somerc +lat_0=47.14439372222222 +lon_0=19.04857177777778 +k_0=0.99993 +x_0=650000 +y_0=200000 +ellps=GRS67 +nadgrids=' + nadgrids_EOV2014 + ' +geoidgrids=' + geoidgrids_EOV2014fine + ' +units=m +no_defs',
                   'EOV2009': '+proj=somerc +lat_0=47.14439372222222 +lon_0=19.04857177777778 +k_0=0.99993 +x_0=650000 +y_0=200000 +ellps=GRS67 +nadgrids=' + nadgrids_EOV2009 + ' +geoidgrids=' + geoidgrids_EOV2009 + ' +units=m +no_defs',

As you can see I am using projection strings.

**Comment by KAMI911** [comment]
Maybe the default preserve_units=True can cause thus problem?
Proj('+proj=geocent +ellps=WGS84 +datum=WGS84 +units=m +no_defs', preserve_units=True)
When I try to force False with:
        self.__SourceProj = Proj(self.__SourceProjectionString, preserve_units=False)
The preserve_units remains True.
Can you check this?

**Comment by snowman2** [comment]
@KAMI911, that should be okay. The `preserve_units=False` just means that on load, convert any `ft` units to `m`. Once the conversion is complete, then to re-create it, the `preserve_units=True` with the modified string is correct.

Would you mind providing examples in code of what you are expecting versus what is actually happening?

**Comment by snowman2** [comment]
Input:
```python
gridetrs = "/home/snowal/projects/pyproj/etrs2eov_notowgs.gsb"
gridgeoid = "/home/snowal/projects/pyproj/geoid_eht.gtx"
from pyproj import Proj, transform
proj1 = Proj('+proj=geocent +ellps=WGS84 +datum=WGS84 +units=m +no_defs ', preserve_units=True)
proj2 = Proj(f'+proj=somerc +lat_0=47.14439372222222 +lon_0=19.04857177777778 +k_0=0.99993 +x_0=650000 +y_0=200000 +ellps=GRS67 +nadgrids={gridetrs} +geoidgrids={gridgeoid} +units=m +no_defs ', preserve_units=True)
```
Output:
```python
# version 1.9.6
>>> transform(proj1, proj2, 4072352.038, 1406781.664, 4687388.295)
(650742.19224405, 251070.04809174273, 123.85061213087161)
# version 2.1.0
>>> transform(proj1, proj2, 4072352.038, 1406781.664, 4687388.295)
(650657.5504410468, 251040.20088167512, 167.514173919335)
```

They seem very close, I wonder if it has to do with something that change in PROJ 6.0.0.

**Comment by snowman2** [comment]
I see you use `somerc` in your projections. I wonder if it is related to https://github.com/OSGeo/proj.4/pull/1234

**Comment by snowman2** [comment]
@kbevers @roualt, can you verify that this issue is related to a change in the code for PROJ?

This is essentially a `proj_create_crs_to_crs`  & `proj_trans_generic` operation.

**Comment by snowman2** [comment]
With PROJ 6.0.0:
```bash
$ cs2cs '+proj=geocent +ellps=WGS84 +datum=WGS84 +units=m +no_defs' '+proj=somerc +lat_0=47.14439372222222 +lon_0=19.04857177777778 +k_0=0.99993 +x_0=650000 +y_0=200000 +ellps=GRS67 +nadgrids=/home/snowal/projects/pyproj/etrs2eov_notowgs.gsb +geoidgrids=/home/snowal/projects/pyproj/geoid_eht.gtx +units=m +no_defs'
4072352.038 1406781.664 4687388.295      
650657.55	251040.20 167.51
```

**Comment by KAMI911** [comment]
@snowman2: Yeah, you are right, it seems there small differences between the transformed coordinates. These are differences (in one example las file):

`Xmin/max/avg/std: -84.6430/-84.6370/-84.6403/0.0014, Ymin/max/avg/std: -29.8510/-29.8460/-29.8485/0.0009, Zmin/max/avg/std: 43.6470/43.6650/43.6579/0.0029`

**Comment by KAMI911** [comment]
Is this fix will be available in may when proj4 will be available with version 6.0.1?

**Comment by KAMI911** [comment]
Is this fix will be available in may when proj4 will be available with version 6.0.1?

**Comment by snowman2** [comment]
I assume so, I think it has been merged in, so it should be included in the next release.

**Comment by snowman2** [comment]
Confirming this has been fixed in PROJ 6.1.0.
```
>>> gridetrs = "/home/snowal/projects/pyproj/etrs2eov_notowgs.gsb"
>>> gridgeoid = "/home/snowal/projects/pyproj/geoid_eht.gtx"
>>> from pyproj import Proj, transform
>>> proj1 = Proj('+proj=geocent +ellps=WGS84 +datum=WGS84 +units=m +no_defs ', preserve_units=True)
>>> proj2 = Proj(f'+proj=somerc +lat_0=47.14439372222222 +lon_0=19.04857177777778 +k_0=0.99993 +x_0=650000 +y_0=200000 +ellps=GRS67 +nadgrids={gridetrs} +geoidgrids={gridgeoid} +units=m +no_defs ', preserve_units=True)
>>> transform(proj1, proj2, 4072352.038, 1406781.664, 4687388.295)
(650742.19224405, 251070.04809174273, 123.84865860538136)
```
